src/Soil.jl

```diff
@@ -1,41 +1,130 @@
 module Soil
+
 using ClimaLSM
 using ClimaCore
 import ClimaCore: Fields, Operators, Geometry
-using ClimaLSM.Domains
-import ClimaLSM:
-    AbstractModel,
-    initialize,
-    initialize_auxiliary,
-    initialize_prognostic,
-    make_update_aux,
-    make_rhs,
-    make_ode_function,
-    prognostic_vars,
-    auxiliary_vars
-export RichardsModel
+import ClimaLSM.Domains: coordinates
+import ClimaLSM: AbstractModel, make_update_aux, make_rhs, prognostic_vars
+using UnPack
+export RichardsModel, RichardsParameters
 
+# Model for Richards equation alone, with only ϑ_l
+# Store everything necessary to define the eqs
 """
     RichardsModel
 
-A standin for a model used to simulate water flow in soil via Richards equation.
+A model for simulating the flow of water in a soil column.
 """
-struct RichardsModel{FT, PS, D} <: AbstractModel{FT}
+struct RichardsModel{FT, PS, D, C, B} <: AbstractModel{FT}
     param_set::PS
     domain::D
+    coordinates::C
+    boundary_exchanges::B
     model_name::Symbol
 end
 
-function RichardsModel{FT}(; param_set, domain::AbstractDomain{FT}) where {FT}
-    return RichardsModel{FT, typeof(param_set), typeof(domain)}(
+function RichardsModel{FT}(; param_set, domain, boundary_exchanges) where {FT}
+    coords = coordinates(domain)
+    args = (param_set, domain, coords, boundary_exchanges)
+    RichardsModel{FT, typeof.(args)...}(
         param_set,
         domain,
+        coords,
+        boundary_exchanges,
         :soil,
     )
 end
 
-prognostic_vars(model::RichardsModel) = (:ϑ_l,)
+coordinates(model::RichardsModel) = model.coordinates
```

Comment:
awesome! this + the model constructor nicely prevents the user from calling coordinates(domain::Column) multiple times and ending up with coordinates from different spaces.